src/dotnet/CarbonAware.LocationSources.Azure/src/AzureLocationSource.cs

```diff
@@ -0,0 +1,113 @@
+using System.Globalization;
+using System.Reflection;
+using CarbonAware.Model;
+using CarbonAware.Interfaces;
+using CarbonAware.Exceptions;
+using Microsoft.Extensions.Logging;
+using System.Text.Json;
+using CarbonAware.LocationSources.Azure;
+
+namespace CarbonAware.LocationSources.Azure;
+
+/// <summary>
+/// Reprsents an azure location source.
+/// </summary>
+public class AzureLocationSource : ILocationSource
+{
+    public string Name => "Azure Location Source";
+
+    public string Description => "Location source that knows how to get and work with Azure location information.";
+
+    private readonly ILogger<AzureLocationSource> _logger;
+
+    private IDictionary<string, NamedGeoposition> namedGeopositions;
+
+    private static readonly JsonSerializerOptions options = new JsonSerializerOptions(JsonSerializerDefaults.Web);
+
+    /// <summary>
+    /// Creates a new instance of the <see cref="AzureLocationSource"/> class.
+    /// </summary>
+    /// <param name="logger">The logger for the datasource</param>
+    public AzureLocationSource(ILogger<AzureLocationSource> logger)
+    {
+        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        namedGeopositions = new Dictionary<string, NamedGeoposition>();
+    }
+
+    public async Task<Location> ToGeopositionLocationAsync(Location location)
+    {
+        switch (location.LocationType)
+        {
+            case LocationType.Geoposition:
+            {
+                return location;
+            }
+            case LocationType.CloudProvider: 
+            {
+                if (location.CloudProvider != CloudProvider.Azure ) 
+                {
+                    throw new LocationConversionException($"Incorrect Cloud provider region. Expected Azure but found '{ location.CloudProvider }'");
+                }
+                var geoPositionLocation = await GetGeoPositionLocationOrThrowAsync(location);
+                return await Task.FromResult(geoPositionLocation);
+            }
+        }
+        
+        throw new LocationConversionException($"Location '{ location.CloudProvider }' cannot be converted to Geoposition. ");
+    }
+
+    private Task<Location> GetGeoPositionLocationOrThrowAsync(Location location)
+    {
+        LoadRegionsFromFileIfNotPresentAsync();
```

Comment:
Having this in the constructor is not a good practice given the fact that it would 'slow' down bootstrap time and it might introduce risk of having to deal with D.I exceptions when the service starts in case something goes wrong (i.e file not found). Better to have this loaded at the 1st request, rather than at startup time.